velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -90,12 +90,90 @@ struct ArrayMinMaxFunction {
   }
 };
 
+template <typename TExecCtx, bool isMax>
+struct ArrayMinMaxFloatFunction {
```

Comment:
It feels like there is a lot of copy-paste. I wonder if we can just add some conditional logic to the existing function.

```
if constexpr (std::is_same_v<T, float> || std::is_same_v<T, double>) {
   bool hasNull = false;
   
   for (auto it = array.begin(); it != array.end(); ++it) {
      if (!it->has_value()) {
           hasNull = true;
      } else if (std::isnan(it->value())) {
          out = it->value();
          return true;
      } 
   }

   if (hasNull) {
      return false;
   }

    return processNoNulls(out, array);
}
```

Here, processNoNulls is a "Input array does not have nulls." loop.

Reply:
Thanks @mbasmanova. The issue here is this approach will require two passes on arrays that does not have any NULL or Nan values. The first loop itself runs through the array once to see if there is a NULL. If not, regular `processNoNulls` will trigger which will need to run through the array again.

The procedure I added do just one pass to the array - hence its a bit different than existing code path. The new procedure is also very clear. Even though it looks similar with the current ArrayMinMax, there are subtle differences (e.g. keep tracking NULL, Nan and min/max) all in one pass.